src/scripts/install_machine.py

```diff
@@ -0,0 +1,326 @@
+#! /usr/bin/python3
+
+import traceback
+import json
+import subprocess
+import sys
+import os
+import xml.etree.ElementTree as ET
+import tempfile
+import logging
+
+
+def virsh(connection, *args):
+    cmd = ("virsh", "-c", f"qemu:///{connection}") + args
+    logging.debug("Running virsh command :" + ' '.join(cmd))
+    return subprocess.check_output(cmd)
+
+
+def assert_vm_exists(connection, name):
+    logging.debug(virsh(connection, 'list', '--all', '--name').strip().split())
+    assert name.encode() in virsh(connection, 'list', '--all', '--name').strip().split()
```

Comment:
Don't call the same command twice, save its output into a variable. Also, this kind of substring match is brittle -- if you ask for "test1" it would be true if only "xxtest12" exists. How about calling `virsh domuuid name` and checking for 0 vs. non-zero exit code? (You'd need to handle `CalledProcessError`, but (1) that's a lot more reliable, and (2) you can certainly create another helper function for calls which are legit to fail)

Also, eventually we should stop calling `virsh` from Python -- we can count on python3-libvirt (as it's a dependency of virt-manager), so using the proper API should be a lot nicer than this shell parsing stuff. However, as you are already that far, let's do this some other time.

Reply:
Applied the domuuid fix now, will change to python API, and maybe we can remove completely the libvirt-client  dependency like that.